addon/components/au-icon.gts

```diff
@@ -39,16 +39,30 @@ export default class AuIcon extends Component<AuIconSignature> {
     }
   }
 
+  get iconComponent() {
+    return typeof this.args.icon !== 'string' && this.args.icon;
+  }
+
   <template>
-    <svg
-      role="img"
-      class="au-c-icon au-c-icon--{{@icon}} {{this.alignment}} {{this.size}}"
-      aria-hidden={{this.ariaHidden}}
-      ...attributes
-    >
-      <use
-        xlink:href="{{this.iconPrefix}}@appuniversum/ember-appuniversum/appuniversum-symbolset.svg#{{@icon}}"
-      ></use>
-    </svg>
+    {{#if this.iconComponent}}
+      {{#let this.iconComponent as |IconComponent|}}
+        {{! @glint-expect-error: glint seems to think that this is not a valid component type  }}
+        <IconComponent
+          class="au-c-icon {{this.alignment}} {{this.size}}"
+          aria-hidden={{this.ariaHidden}}
```

Comment:
I'm not sure if we should pass any extra attributes to this component actually. This feels like we leak implementation details of the `AuIcon` component into the custom ones? I think we should just render the custom component as-is for now.

Reply:
Hmm, I was wondering about that but also wanted to take care of accessibility, as well as preventing the need to manually match the styling and sizing. I guess we could just document the classes/rules to apply to match the existing implementation and advice on accessibility? Or we could put the Icon in a div, but I worry that will have other side-effects.

Reply:
Hmm, yea, maybe. I just wonder if there might be cases where we don't want these attributes. I'm probably overthinking and this might be the safest option. 

I do still wonder why we would ever set aria-hidden to something else than "true" though, I'm not sure why this was ever an argument in the first place. Do you think it makes sense to always set this to true for the icon components, since it's an opt-in?

Reply:
So you're happy with leaving these attributes? I'm wondering if we should give the option to opt out of these attributes being applied.

Reply:
> So you're happy with leaving these attributes?

Yea, it seems like the safest option to start with 👍.

> I'm wondering if we should give the option to opt out of these attributes being applied.

Well, yea that's what I was originally thinking, but that's probably not needed. I suspect that most consumers will always use one of the bundled icon components, or a custom svg component. I don't expect that people will want to fully replace the AuIcon behavior, but I could be wrong.

I think this is one of those cases that we can't predict, and only encounter once we start using this system in our apps. So I would keep it simple for now.